src/execution/syscall_handler.rs

```diff
@@ -45,6 +45,18 @@ impl OsSyscallHandlerWrapper {
         self.syscall_handler.as_ref().borrow().syscall_ptr
     }
 
+    pub fn validate_and_discard_syscall_ptr(&self, syscall_ptr_end: Relocatable) -> Result<(), HintError> {
```

Comment:
It is implemented in my pr: https://github.com/keep-starknet-strange/snos/blob/7b9f2a3c2985277b779df5c9249d40f871c4cc33/src/execution/syscall_handler.rs#L48

Reply:
Okay so I see you modified the code to make `syscall_ptr` an `Option<Relocatable>`. I don't know how big your PR will end up being, what do you think about fixing and merging this PR?

Reply:
Makes sense.